tweetflask.py

```diff
@@ -24,41 +29,66 @@ def get():
     # looking up the username in the DB
     for user in jsondata['UserData']:
         if username.upper().lower() == user['Username'].upper().lower():
-            return jsonify(get_tweet(jsondata, user, mood))
+            return jsonify(get_tweet(jsondata, user, mood, None))
     # if the username does not exist in DB this line will be executed
     # to return an empty array
-    return jsonify(get_tweet(jsondata, None, None))
+    return jsonify(get_tweet(jsondata, None, None, None))
+
+
+@app.route('/tweets/<get_id>')
+def get_ID(get_id):
+    # Opening and reading the db file
+    jsondata = read_file()
+
+    for tweets in jsondata['TweetData']:
+        if get_id == tweets['TweetID']:
+            for user in jsondata['UserData']:
+                if tweets['UserID'] == user['UserID']:
```

Comment:
The logic is repeated with [here](https://github.com/osu-mist/tweet-api/pull/7/files#diff-0ba64d63ad493d9524f55fc910a3ffd4R62). I can tell you were trying to iterate all users and tweets to get the correct result, but it's a bit hard to read/understand the code with multiple nested loops/if-else statements. I'd suggest creating a separate function/logic (like you did for the `read_file` function) to find the corresponding user from tweets to simplify the code.

In addition, since you only have very few users and tweets in your database, it's okay to iterate all data to fetch the result. However, how about if you have 100,000+ users and 1,000,000 tweets in your database? We definitely don't wanna to loop through all of them every time. Can you think of any way to improve the performance?

Reply:
I can search by doing tweet = jsondata['TweetData'][int(tweet_id)], but I was thinking what if someone deleted a tweet, then the tweets id's will not be in order.
is there a way to access json property by value instead of index number?

Reply:
i'm looking into objectpath